cmake/cmake_tools.cmake

```diff
@@ -0,0 +1,187 @@
+mark_as_advanced(DEFAULT_CLANG_TIDY_ERROR_ARGS)
+
+# Adds clang-tidy checks to the target, with the given arguments being used
+# as the options set.
+macro(target_clang_tidy target)
+  if(CLANG_TIDY_EXE)
+    get_target_property(${target}_type ${target} TYPE)
+    if (NOT ${${target}_type} STREQUAL "INTERFACE_LIBRARY")
+      set_target_properties("${target}" PROPERTIES CXX_CLANG_TIDY "${CLANG_TIDY_EXE};${ARGN}")
+    endif()
+  endif()
+endmacro()
+
+# Adds include_what_you_use to the target, with the given arguments being
+# used as the options set.
+macro(target_include_what_you_use target)
+  if(IWYU_EXE)
+    set_target_properties("${target}" PROPERTIES CXX_INCLUDE_WHAT_YOU_USE "${IWYU_EXE};${ARGN}")
+  endif()
+endmacro()
+
+# Adds include_what_you_use to all targets, with the given arguments being used as the options set.
+macro(include_what_you_use)
+  if(IWYU_EXE)
+    set(CMAKE_CXX_INCLUDE_WHAT_YOU_USE "${IWYU_EXE};${ARGN}")
+  endif()
+endmacro()
+
+# Adds cppcheck to the target, with the given arguments being used as the options set.
+macro(target_cppcheck target)
+  if(CPPCHECK_EXE)
+    set_target_properties("${target}" PROPERTIES CXX_CPPCHECK "${CPPCHECK_EXE};${ARGN}")
+  endif()
+endmacro()
+
+# Adds cppcheck to all targets, with the given arguments being used as the options set.
+macro(cppcheck)
+  if(CPPCHECK_EXE)
+    set(CMAKE_CXX_CLANG_TIDY "${CPPCHECK_EXE};${ARGN}")
+  endif()
+endmacro()
+
+# Performs multiple operation so other packages may find a package
+# Usage: configure_package(NAMSPACE namespace TARGETS targetA targetb)
+#   * It installs the provided targets
+#   * It exports the provided targets under the provided namespace
+#   * It installs the package.xml file
+#   * It create and install the ${PROJECT_NAME}-config.cmake and ${PROJECT_NAME}-config-version.cmake
```

Comment:
Would it be useful for use to export a CMake variable that contains all of the targets in the project? Or does that already happen? I'm thinking ${PACKAGE_NAME}_TARGETS. 

Along those lines, do we have an example cmake.in file?

Reply:
I am not sure it is already happening but we can open and issue as an enhancement.